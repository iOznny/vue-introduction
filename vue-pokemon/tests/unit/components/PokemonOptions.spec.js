import { shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions';
import { mockPokemons } from '../mocks/pokemons.mock';

describe('Component - Pokemon Options', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: mockPokemons
            }
        });
    })
    
    test('Match with snapshot', () => {        
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Show fore options successful', () => {
        const liTags = wrapper.findAll('li');

        // Find four <li>
        expect(liTags.length).toBe(4);

        // Contains pokemon
        expect(liTags[0].text()).toBe('PIKACHU');
        expect(liTags[1].text()).toBe('CHARMANDER');
        expect(liTags[2].text()).toBe('CHARIZARD');
        expect(liTags[3].text()).toBe('MEW');
    })

    test('Emit "selection" with params', () => {
        const [li1, li2, li3, li4] = wrapper.findAll('li');

        // Emit event 'click'
        li1.trigger('click');
        li2.trigger('click');
        li3.trigger('click');
        li4.trigger('click');

        // Hope 4 emits
        expect(wrapper.emitted('selection-pokemon').length).toBe(4);

        // Checking li with id 
        expect(wrapper.emitted('selection-pokemon')[0]).toEqual([1]);
        expect(wrapper.emitted('selection-pokemon')[1]).toEqual([2]);
        expect(wrapper.emitted('selection-pokemon')[2]).toEqual([3]);
        expect(wrapper.emitted('selection-pokemon')[3]).toEqual([4]);
    })
})