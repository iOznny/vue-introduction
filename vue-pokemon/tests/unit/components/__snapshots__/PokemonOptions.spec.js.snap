// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Pokemon Options Match with snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>PIKACHU</li>
    <li>CHARMANDER</li>
    <li>CHARIZARD</li>
    <li>MEW</li>
  </ul>
</div>
`;
